test/eventually_test.go

```diff
@@ -0,0 +1,145 @@
+package test
+
+import (
+	"github.com/orbs-network/go-mock"
+	"testing"
+	"time"
+)
+
+func TestEventually(t *testing.T) {
+	num := 1
+	go func() {
+		time.Sleep(1 * time.Millisecond)
+		num = 2
+	}()
+	ok := Eventually(func() bool {
+		return num == 2
+	})
+	if !ok {
+		t.Fatal("Eventually did not discover change to 2")
+	}
+}
+
+func TestConsistently(t *testing.T) {
+	num := 1
+	go func() {
+		time.Sleep(1 * time.Millisecond)
+		num = 2
+	}()
+	ok := Consistently(func() bool {
+		return num == 1
+	})
+	if ok {
+		t.Fatal("Consistently did not discover change to 2")
+	}
+}
+
+type personMock struct {
+	mock.Mock
+}
+
+func (p *personMock) GetName() string {
+	return p.Called().String(0)
+}
+
+func TestEventuallyVerifySuccess(t *testing.T) {
+	p := &personMock{}
+	p.When("GetName").Return("john").Times(1)
+	go func() {
+		time.Sleep(1 * time.Millisecond)
+		p.GetName()
+	}()
+	err := EventuallyVerify(p)
+	if err != nil {
+		t.Fatal("EventuallyVerify did not discover mock was eventually called")
+	}
+}
+
+func TestEventuallyVerifyFailure(t *testing.T) {
+	p := &personMock{}
+	p.When("GetName").Return("john").Times(1)
+	err := EventuallyVerify(p)
+	if err == nil {
+		t.Fatal("EventuallyVerify did not discover mock was not eventually called")
+	}
+}
+
+func TestEventuallyVerifySuccessWithTwoMocks(t *testing.T) {
+	p1 := &personMock{}
+	p1.When("GetName").Return("john").Times(1)
+	p2 := &personMock{}
+	p2.When("GetName").Return("smith").Times(1)
+	go func() {
+		time.Sleep(1 * time.Millisecond)
```

Comment:
consider controlling the flow in this test using stuff in the `sync` package rather than using sleeps, as this is fragile

Reply:
I'll try

Reply:
I only succeeded removing the sleeps from the first two tests. I wasn't successful in removing them from the `EventuallyVerify` and `ConsistentlyVerify` tests that work with mocks

Reply:
let's review together

Reply:
separately from this PR though so it doesn't get stuck